### PR TITLE
feat: add analytics events to search

### PR DESCRIPTION
**Pull request by gabriel-dantas98**
## What & Why

We would like to have more data on the qeta search so that we can better understand how people are using our engineering.

I was inspired by the use of the other analytics calls in the plugin.

### CODE REVIEW

File: plugins/qeta/src/components/QuestionsContainer/QuestionsContainer.tsx [hunk 1]
```diff
@@ -87,6 +89,7 @@ export const QuestionsContainer = (props: QuestionsContainerProps) => {
 
   const onSearchQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
     onPageChange(1);
+    analytics.captureEvent('qeta-search', event.target.value ?? '');
```
Comment (gabriel-dantas98) [inline plugins/qeta/src/components/QuestionsContainer/QuestionsContainer.tsx:92]:
```suggestion
    analytics.captureEvent('qeta-search', event.target.value ?? '');
```
I used _qeta-_ as a prefix so as not to confuse it with the native backstage search

Comment (gabriel-dantas98) [inline plugins/qeta/src/components/QuestionsContainer/QuestionsContainer.tsx:92]:
```suggestion
    analytics.captureEvent('search', event.target.value ?? '');
```

File: plugins/qeta/src/components/QuestionsContainer/QuestionsContainer.tsx [hunk 2]
```diff
@@ -87,6 +89,7 @@ export const QuestionsContainer = (props: QuestionsContainerProps) => {
 
   const onSearchQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
     onPageChange(1);
+    analytics.captureEvent('search', event.target.value ?? '');
```
Comment (drodil) [inline plugins/qeta/src/components/QuestionsContainer/QuestionsContainer.tsx:92]: The backstage search uses analytics with the same scope here: https://github.com/backstage/backstage/blob/master/plugins%2Fsearch-react%2Fsrc%2Fcontext%2FSearchContext.tsx#L141-L142

We don't want to mix Q&A analytics with that so I suggest changing this to `qeta_search` or something similar
 Also I suggest that you check the value is not empty or undefined before capturing event (like backstage search does)

**Comment by drodil** [comment]
Thanks for this, it makes sense! Just few points in the comment to think about!

**Pull request merged**